internal/pkg/mining/scheduler.go

```diff
@@ -7,108 +7,127 @@ package mining
 
 import (
 	"context"
+	"fmt"
 	"sync"
 	"time"
 
 	"github.com/filecoin-project/go-filecoin/internal/pkg/block"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/clock"
+	th "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers"
 	"github.com/pkg/errors"
 )
 
-// Scheduler is the mining interface consumers use. When you Start() the
-// scheduler it returns a channel and a sync.WaitGroup:
-//   - channel: the scheduler emits outputs (mined blocks) to the caller on this channel
-//   - waitgroup: signals that the scheduler and any goroutines it launched
-//             have stopped. (Context cancelation happens async, so you
-//             need some way to know when it has actually stopped.)
-//
-// Once started, a scheduler can be stopped by canceling the context provided to `Start()`,
-// which will signal on the waitgroup when it's actually done.
+// Scheduler is the mining interface consumers use.
 type Scheduler interface {
-	Start(miningCtx context.Context) (<-chan Output, *sync.WaitGroup)
+	Start(miningCtx context.Context, skipCh chan bool) (<-chan Output, *sync.WaitGroup)
 	IsStarted() bool
```

Comment:
given that mining can be paused/skipped now, we might want to add a `IsMining() bool`

Reply:
Also not a bad idea, though I might wait until we have a reason to surface that info.

Reply:
Decided to defer for now